packages/taco/test/taco-sign.test.ts

```diff
@@ -0,0 +1,214 @@
+
+      signUserOpMock.mockResolvedValue({
+        digest: '0x1234',
+        aggregatedSignature: '0x90ab0xijkl',
+        signingResults: {
+          '0x1234': ['0x5678', '0x90ab'],
+          '0xabcd': ['0xefgh', '0xijkl'],
+        },
+        errors: {},
+      });
+
+      const result = await signUserOp(
+        mockProvider,
+        userOp,
+        chainId,
+        aaVersion,
+        cohortId,
+        'lynx',
+        { optimistic: true, returnAggregated: true },
+        undefined,
+        porterUris
+      );
+
+      expect(signUserOpMock).toHaveBeenCalledWith(
+        {
+          '0x1234': JSON.stringify({
+            userOp,
+            cohortId,
+            chainId,
+            aaVersion,
+            context: undefined
+          }),
+          '0xabcd': JSON.stringify({
+            userOp,
+            cohortId,
+            chainId,
+            aaVersion,
+            context: undefined
+          })
+        },
+        2,
+        { optimistic: true, returnAggregated: true }
+      );
+
+      expect(result).toEqual({
+        digest: '0x1234',
+        aggregatedSignature: '0x90ab0xijkl',
+        signingResults: {
+          '0x1234': ['0x5678', '0x90ab'],
+          '0xabcd': ['0xefgh', '0xijkl'],
+        },
+        errors: {},
+      });
+    });
+
+    it('should sign a user operation with kernel account', async () => {
+      const chainId = 1;
+      const aaVersion = 'kernel:v0.7';
+      const cohortId = 5;
+      const porterUris = [fakePorterUri];
+
+      signUserOpMock.mockResolvedValue({
+        digest: '0x1234',
+        aggregatedSignature: '0x90ab0xijkl',
+        signingResults: {
+          '0x1234': ['0x5678', '0x90ab'],
+          '0xabcd': ['0xefgh', '0xijkl'],
+        },
+        errors: {},
+      });
+
+      const result = await signUserOp(
+        mockProvider,
+        userOp,
+        chainId,
+        aaVersion,
+        cohortId,
+        'lynx',
+        { optimistic: true, returnAggregated: true },
+        undefined,
+        porterUris
+      );
+
+      expect(signUserOpMock).toHaveBeenCalledWith(
+        {
+          '0x1234': JSON.stringify({
+            userOp,
+            cohortId,
+            chainId,
+            aaVersion,
+            context: undefined
+          }),
+          '0xabcd': JSON.stringify({
+            userOp,
+            cohortId,
+            chainId,
+            aaVersion,
+            context: undefined
+          })
+        },
+        2,
+        { optimistic: true, returnAggregated: true }
+      );
+
+      expect(result).toEqual({
+        digest: '0x1234',
+        aggregatedSignature: '0x90ab0xijkl',
+        signingResults: {
+          '0x1234': ['0x5678', '0x90ab'],
+          '0xabcd': ['0xefgh', '0xijkl'],
+        },
+        errors: {},
+      });
+    });
+
+    it('should handle custom signing options', async () => {
+      const chainId = 1;
+      const aaVersion = 'safe:v0.8';
```

Comment:
Let's use currently valid types:

```suggestion
      const aaVersion = '0.8.0';
```